.github/workflows/integration_tests.yml

```diff
@@ -0,0 +1,106 @@
+name: Integration tests
+on:
+  pull_request:
+    paths-ignore:
+    - '*.md'
+    - '**/*.md'
+  push:
+    paths-ignore:
+    - '*.md'
+    - '**/*.md'
+    branches:
+    - main
+permissions:
+  contents: read
+env:
+  GH_ANNOTATION: true
+jobs:
+  docker_build:
+    runs-on: ubuntu-20.04
+    name: Build SMI Extension
+    timeout-minutes: 30
+    steps:
+    - name: Checkout code
+      # actions/checkout@v2
+      uses: actions/checkout@722adc63f1aa60a57ec37892e133b1d319cae598
+    - name: Set environment variables from scripts
+      run: |
+        . bin/_tag.sh
+        echo "TAG=$(CI_FORCE_CLEAN=1 bin/root-tag)" >> $GITHUB_ENV
+
+        echo "DOCKER_REGISTRY=cr.l5d.io/linkerd" >> $GITHUB_ENV
+        echo "DOCKER_BUILDKIT_CACHE=${{ runner.temp }}/.buildx-cache" >> $GITHUB_ENV
+    - name: Cache docker layers
+      # actions/cache@v2.0.0
+      uses: actions/cache@b8204782bbb5f872091ecc5eb9cb7d004e35b1fa
+      with:
+        path: ${{ env.DOCKER_BUILDKIT_CACHE }}
+        key: ${{ runner.os }}-buildx-smi-${{ env.TAG }}
+        restore-keys: |
+          ${{ runner.os }}-buildx-smi-
+    - name: Build SMI docker images and CLI
+      run: |
+        docker buildx create --driver docker-container --use
+        bin/docker-build
+        bin/build-cli-bin
+    - name: Create artifact with CLI and image archives
+      env:
+        ARCHIVES: /home/runner/archives
+      run: |
+        mkdir -p $ARCHIVES
+        docker save "cr.l5d.io/linkerd/smi-adaptor:$TAG" > $ARCHIVES/smi-adaptor.tar
+        cp target/cli/linux-amd64/linkerd-smi $ARCHIVES
+    - name: Upload artifact
+      # actions/upload-artifact@v1
+      uses: actions/upload-artifact@3446296876d12d4e3a0f3145a3c87e67bf0a16b5
+      with:
+        name: build-archives
+        path: /home/runner/archives
+  integration_tests:
+    needs: [docker_build]
+    runs-on: ubuntu-20.04
+    timeout-minutes: 60
+    strategy:
+      matrix:
+        # TODO: Test on stable version too
+        version: [ edge]
```

Comment:
This is not needed as its very hard to add support in stable, because the SP validator actively rejects SPs with empty `len(sp.Spec.Routes)`. So, Removing this strategy stuff altogether.